### PR TITLE
Feature/modal

### PR DESCRIPTION
**Pull request by StrongCurrent**
Now a Modal is integrated with the delete spot and delete info button. Happy about feedback. THX 🙏

### CODE REVIEW

File: components/SpotInfo/style.js [hunk 1]
```diff
@@ -117,3 +117,57 @@ export const SpotNameChangedSuccess = styled.div`
...
+  font-weight: 700;
+  color: #ff0000;
+  margin:0px 0px 10px 0px;
+`;
+
+export const ModalMessage = styled.p`
+  font-size: 1rem;
+  font-weight: 500;
+  color: #000000;
+  margin-top: 0px;
```
Comment (onemanwenttomow) [inline components/SpotInfo/style.js:133]: Minor thing, but if the value if 0, then we do not need the "px". This is generally down to personal preference though.
```suggestion
  margin-top: 0;
```

File: components/EditDeleteInfoForm/index.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect, useCallback } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (onemanwenttomow) [inline components/EditDeleteInfoForm/index.js:1]: You don't seem to be using ```React``` in this component (you no longer need to import) so i think we can remove that. 

```suggestion
import { useState, useEffect } from "react";
```

File: components/EditDeleteInfoForm/index.js [hunk 3]
```diff
@@ -134,7 +133,7 @@ export default function EditDeleteInfoForm({ entry, spotId }) {
                 )}
               </EntryEditButton>
               <EntryDeleteButton
-                onClick={handleDelete}
+                onClick={openModal}
                 aria-label={`Delete entry for ${entry.info}`}
                 disabled={disabled}
```
Comment (onemanwenttomow) [inline components/EditDeleteInfoForm/index.js:138]: Minor thing, and not requesting any changes but... the exact appearance of disabled buttons can vary slightly between different browsers and versions. If you want to ensure a consistent and specific visual style for disabled buttons across different browsers, you can use CSS to customise the styling

File: components/EditDeleteInfoForm/style.js [hunk 4]
```diff
@@ -92,3 +92,57 @@ export const EntryDeleteButton = styled.button`
...
+`;
+
+export const ModalMessage = styled.p`
+  font-size: 1rem;
+  font-weight: 500;
+  color: #000000;
+  margin-top: 0px;
+`;
+
+export const ModalDeleteButton = styled.button`
```
Comment (onemanwenttomow) [inline components/EditDeleteInfoForm/style.js:111]: You might want to consider having as base style for your buttons. And then creating specific button styles based off of that. 

[See example here](https://styled-components.com/docs/basics#extending-styles).

The following code seems to be shared by both the `ModalDeleteButton` and the `ModalKeepButton`

``` css
text-transform: uppercase;
color: #ffffff;
height: 25px;
width: 120px;
border-radius: 5px;
border: none;

&:focus {
  font-weight: 600;
}
```

Having one base button will make maintaining things easier in the future.

File: components/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
```
Comment (onemanwenttomow) [inline components/Modal/index.js:1]: I don't think we need this line. 

```suggestion

```

File: components/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    <ModalOverlay>
+      <ModalWrapper>
+        <StyledModal>
+          <ModalBody>{children}</ModalBody>
+        </StyledModal>
+      </ModalWrapper>
+    </ModalOverlay>
+  );
+
+  return ReactDOM.createPortal(
```
Comment (onemanwenttomow) [inline components/Modal/index.js:26]: If we destructure above, then we can do the following here.

```suggestion
  return createPortal(
```

File: components/Modal/index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import ReactDOM from "react-dom";
```
Comment (onemanwenttomow) [inline components/Modal/index.js:2]: We could destructure here [(see another example in react docs)](https://react.dev/reference/react-dom/createPortal#reference)

```suggestion
import { createPortal } from "react-dom";
```

File: components/SpotInfo/index.js [hunk 8]
```diff
@@ -126,13 +137,13 @@ export default function SpotInfo({ spotId }) {
           aria-label="Edit spot name"
         >
           {isEditingSpotName ? (
-            <EditIcon as={CiCircleCheck} size={25} />
+            <EditIcon as={CiCircleCheck} size={25} aria-hidden="true" />
           ) : (
-            <EditIcon as={CiEdit} size={25} />
+            <EditIcon as={CiEdit} size={25} aria-hidden="true" />
           )}
```
Comment (onemanwenttomow) [inline components/SpotInfo/index.js:143]: We could shorten this a little by moving the ternary to the as prop. i.e. 

``` js
<EditIcon as={isEditingSpotName ? CiCircleCheck : CiEdit} size={25} aria-hidden="true" />

```

File: components/SpotInfo/index.js [hunk 9]
```diff
@@ -1,9 +1,10 @@
-import React, { useState, useCallback } from "react";
+import React, { useState } from "react";
```
Comment (onemanwenttomow) [inline components/SpotInfo/index.js:1]: Again, i don't think we need to import `React` here. 

> You no longer need to import React from "react". Starting from the release 17 of React

File: components/SpotInfo/index.js [hunk 10]
```diff
@@ -15,32 +16,31 @@ import {
...
-    isValidating,
-    mutate,
-  } = useSWR(`/api/spots/${spotId}`);
+  const { data: spot, error, mutate } = useSWR(`/api/spots/${spotId}`);
   const [isEditingSpotName, setIsEditingSpotName] = useState(false);
   const [spotNameError, setSpotNameError] = useState("");
   const [newSpotName, setNewSpotName] = useState("");
   const [spotNameChangeSuccess, setSpotNameChangeSuccess] = useState(false);
   const router = useRouter();
+  const [isModalOpen, setIsModalOpen] = useState(false);
```
Comment (onemanwenttomow) [inline components/SpotInfo/index.js:37]: Very minor bit of house keeping, but i would group all your useStates together. i.e. 

``` js
const router = useRouter();

const [isEditingSpotName, setIsEditingSpotName] = useState(false);
const [spotNameError, setSpotNameError] = useState("");
const [newSpotName, setNewSpotName] = useState("");
const [spotNameChangeSuccess, setSpotNameChangeSuccess] = useState(false);
const [isModalOpen, setIsModalOpen] = useState(false);
```

**Review by onemanwenttomow — APPROVED** [review]
@StrongCurrent Nice work! I added a few thoughts and comments for you (mostly minor things).

**Comment by StrongCurrent** [comment]
> @StrongCurrent Nice work! I added a few thoughts and comments for you (mostly minor things).

Thank you for your helpful feedback and tips. I have looked at all this again 🥰

**Pull request merged**